common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -1198,6 +1219,16 @@ private static boolean equalsSafe(byte[] bytes1, int startPos1, byte[] bytes2, i
         return true;
     }
 
+    private static boolean equalsSafeConsistentTime(byte[] bytes1, int startPos1,
+                                                    byte[] bytes2, int startPos2, int length) {
+        byte b = 0;
+        final int end = startPos1 + length;
+        for (int i = startPos1, j = startPos2; i < end; ++i, ++j) {
+            b |= bytes1[i] ^ bytes2[j];
+        }
+        return b == 0;
```

Comment:
The [Go](https://golang.org/src/crypto/subtle/constant_time.go?s=490:531#L2) implementation uses a constant time comparison for the final check here.  You might consider using it too.